modules/custom/openy_search/openy_google_search/src/Config/GoogleSearchOverrides.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Drupal\openy_google_search\Config;
+
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Config\ConfigFactory;
+use Drupal\Core\Config\ConfigFactoryOverrideInterface;
+use Drupal\Core\Config\StorageInterface;
+
+/**
+ * OpenY Google Search configuration override.
+ */
+class GoogleSearchOverrides implements ConfigFactoryOverrideInterface {
+
+  /**
+   * Configuration factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactory
+   */
+  private $configFactory;
+
+  /**
+   * Constructs a DomainConfigSubscriber object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactory $config_factory
+   *   Configuration factory.
+   */
+  public function __construct(ConfigFactory $config_factory) {
+    $this->configFactory = $config_factory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function loadOverrides($names) {
+    $default_theme = $this->configFactory->getEditable('system.theme')->getOriginal('default');
+    $overrides = [];
+    if (in_array($default_theme . '.settings', $names)) {
+      $overrides[$default_theme . '.settings'] = ['search_query_key' => 'q'];
+      $overrides[$default_theme . '.settings'] = ['display_search_form' => 1];
```

Comment:
You rewrite the entire config here. The `search_query_key` is lost after `display_search_form` is set, so, in the end, the search input doesn't have the name attribute :-)

Reply:
To reproduce the problem:
- open the search form in the header when the module is installed
- put the value in the search input field
- submit the form
- you are redirected to `/search?` <--- see, no query term in the URL!
- take a look in the page markup and you'll see `<input type="search" name="" class="search-input" placeholder="Search">` <---- see, name is empty!

Reply:
I should also add that these settings should not only apply to a default theme, but to any enabled theme.

Reply:
Why add to all active themes, because among them there is also an administrative theme. It certainly does not need these parameters. Therefore, the default theme that is currently displayed to the user will be enough.

Reply:
Well. I partially agree. But the default theme is not an active theme, right? It is possible that 2 themes are enabled and they are switched with, for example, theme key module.